src/Text/Pandoc/Readers/Docx/Parse.hs

```diff
@@ -236,12 +240,82 @@ newtype TblLook = TblLook {firstRowFormatting::Bool}
 defaultTblLook :: TblLook
 defaultTblLook = TblLook{firstRowFormatting = False}
 
-newtype Row = Row [Cell]
-           deriving Show
+data Row = Row TblHeader [Cell] deriving Show
+
+data TblHeader = HasTblHeader | NoTblHeader deriving (Show, Eq)
 
-newtype Cell = Cell [BodyPart]
+data Cell = Cell GridSpan VMerge [BodyPart]
             deriving Show
 
+type GridSpan = Integer
+
+data VMerge = Continue
+            -- ^ This cell should be merged with the one above it
+            | Restart
+            -- ^ This cell should not be merged with the one above it
+            deriving (Show, Eq)
+
+-- Can I transpose and then transpose again when the table isn't a
+-- perfect grid (e.g. when there are colspans)?
+-- (nope, things will get moved around)
+-- need to rewrite transpose to take the colspans into account
+-- (or come up with a different way of doing this)
+-- rowsToRowspans :: [Row] -> [[Int]]
+-- rowsToRowspans rows = transpose (fmap rowspans (columns rows))
+--   where
+--     columns :: [Row] -> [[Cell]]
+--     columns = transpose . fmap (\(Row _ cs) -> cs)
+
+--     rowspans :: [Cell] -> [Int]
+--     rowspans = fmap snd . fst . foldr calculateRowspan ([], 0)
+
+--     -- TODO: Should the contents of merged cells be combined?
+--     calculateRowspan :: Cell -> ([(Cell, Int)], Int) -> ([(Cell, Int)], Int)
+--     calculateRowspan cell@(Cell _ vMerge _) (column, continuesBelow) =
+--       case vMerge of
+--         Restart -> ((cell, 1 + continuesBelow) : column, 0)
+--         Continue -> ((cell, 0) : column, 1 + continuesBelow)
+
+rowsToRowspans :: [Row] -> [[(Int, Cell)]]
+rowsToRowspans rows = let
```

Comment:
@mb21 I want to unit test this function, but I can't figure out where to put the tests. I've tried `test/Tests/Readers/Docx.hs`, but I can't import the function in that file. Any idea how I can make that work?